File: src/main/java/com/csl/HelloWorld.java
package com.csl;

import com.csl.student.Student;

public class HelloWorld {

    public static void main(String[] args) {

        Person p = new Person();
        p.hello();

        int age =19;
        Integer age2 = 19;
        char c = 'a';
        Character c2 = 'A';
        byte b = 120;
        float weight = 66.5f;
        double height = 175;
        boolean enroll = true;
        String s ="str";


        p.weight = 55.6f;
        p.height = 1.7f;
        System.out.println(p.bmi());

        Person p2 = new Person(55.6f, 1.7f);
        System.out.println(p.bmi());


        int score = 80;
        System.out.println(score>80 && score<90);
        //JAVA中 char昰數值 A=65, a=97
        char c3 = 'A';
        System.out.println(c3 > 60);



    }


    //快速鍵 Help -> keymap reference
    //VCS = alt + `
    //叫出version control = alt + 9
    //複製游標所在的該行到下一行去 = ctrl + D
    //區塊注解 = ctrl + shift + /
    //debug 執行下一行 = F8
    //debug step nito執行下一行 = F7
    //查看方法可設定之參數 = ctrl + p (多載 overloading)
    //注解單行 = crtl +/
    //叫出功能選單 = alt + iinsert
    //抽取出程method = ctrl + alt +M
    //新增Constructor/getter setter = alt + insert
    //查看function可以放什麼種類的參數 = ctrl+P
    //加入try catch = ctrl + atl + t


    //System.out.println = sout + tab
    //public static void main(String[] args)  = psvm + tab

    //八個基本資料型態 bool int short long byte float double char
    //String 是物件不是型態 因為太常使用所以自行呼叫new String()
    // +號 左右兩邊有任一個是String  另一邊若不是String會自動轉成String
    // =號 右邊只能放Expression 不能放 statement(敘述式 if~else)

}

File: src/main/kotlin/com/csl/HelloWorld.kt
package com.csl

//kotlin 和 Java 在同一個package底下能有相同的Class

fun main (args: Array<String>) {
        //println("Hello world")
        //Human().hello()
        val h = Human(weight = 1.0f, height = 0f)

        //var: 值可改變
        //val: 值不可改
        //變數型態自動推斷 / 或自己加入資料型態
        val v1 = 10
        var v2 = 10.5
        v2=9.5;

        var i:Int = 10
        var d:Double = 10.5
        var b:Boolean = true
        var f:Float = 10.1f
        var s:String  = "str"
        var c:Char = 'c';
        var byte:Byte = 12

        var h2 = Human(weight = 55.6f, height = 1.7f)
        println(h2.bmi())


        var h3 = Human("abc", 5f, 10f)

        //kotlon Char不是數值，若要轉成int使用 toInt()
        val score = 88
        println(score>60)
        val c2:Char = 'A'
        println( c2.toInt() > 60)

}

//如果不想讓constructor的參數成為property, 就不要加var/val
class Human(var name:String="jack", var weight:Float, var height:Float) {

        //建構後執行的內容
        init {
                //字串裡要有變數 直接加$
                println("weight = $weight")
        }

        //次要建構子(secondary constructor) 傳入參數不使用var,
        // 且如傳入參數primary constructor已設定，要加屬性-主建構子 this()
        // 且一樣會執行 init()
      //  constructor(name: String, weight: Float, height: Float): this(weight, height)

        fun bmi(): Float {
                var bmi = weight / (height*height)
                return bmi
        }

        fun hello() {
                println("Hello kotlin")
        }
}

// kotlin 沒有三元運算式  因為 =號/return/字串+法 後面可以放if~else~(JAVA if else 是statement Kotlin則變成Expression)
// 字串裡呼叫方法 ${ method }